src/examples/optimized-build-and-push.yml

```diff
@@ -0,0 +1,27 @@
+description: >
+  Log into Google Cloud Plaform, then build and push image to GCR. Uses a custom executor to reduce the setup time.
+
+usage:
+  version: 2.1
+
+  orbs:
+    gcp-gcr: circleci/gcp-gcr@x.y.z
+
+  executors:
+    my-executor:
+      docker:
+        # choose an image that contains the gcloud CLI to avoid installation during CI
+        # you can extend CircleCI base images https://circleci.com/docs/2.0/circleci-images/#circleci-base-image
+        # example: https://github.com/RequestNetwork/circleci/blob/master/python/3.7/Dockerfile
+        - image: requestnetwork/circleci-python:3.7
```

Comment:
I'm not sure we can accept promoting a 3rd party image within our official examples, but the sentiment is understood and we agree. @felicianotech and I have been discussing for some time creating a suite of deployment images which would contain the CLIs. There's probably a few approaches there we should discuss.

This may not block this PR waiting for those images but I feel we may need to limit our examples to only include official docker images and those we control.

Reply:
No worries this was given as an example... Writing it I felt also like this should be an official image from circle CI
If you need I can share the dockerfile so you publish you own ?